samples: matter: Added snippet for schedules to Matter Lock sample (#16710)
nrfconnect/sdk-nrf: samples/matter/lock/README.rst

     - ``Passage`` - The lock can be operated without providing a PIN.
       This option can be used, for example, for employees during working hours.

> To enable the schedule timed access feature use the ``schedules`` snippet.

Reviewer: ```suggestion
To enable the scheduled timed access feature, use the ``schedules`` snippet.
```

> See :ref:`matter_lock_snippets` to learn how to enable a snippet in Matter lock sample.

Reviewer: ```suggestion
See the :ref:`matter_lock_snippets` section of this sample to learn how to enable the snippet.
```

> See :ref:`matter_lock_sample_schedule_testing` sections of this sample for more information about testing the schedule timed access feature.

Reviewer: ```suggestion
See the :ref:`matter_lock_sample_schedule_testing` section of this sample for more information about testing the scheduled timed access feature.
```

.. matter_door_lock_sample_configuration_file_types_end

.. _matter_lock_snippets:

Snippets
========

.. |snippet| replace:: :makevar:`lock_SNIPPET`

.. include:: /includes/sample_snippets.txt

The following snippet is available:

* ``schedules`` - Enables schedule timed access features.

> .. |snippet_zap_file| replace:: :file:`snippets/schedules/lock.zap`
> .. |snippet_dir| replace:: :file:`snippets/schedules`
> 
> .. include:: /includes/matter_snippets_note.txt

Reviewer: ```suggestion
  .. |snippet_zap_file| replace:: :file:`snippets/schedules/lock.zap`
  .. |snippet_dir| replace:: :file:`snippets/schedules`

  .. include:: /includes/matter_snippets_note.txt
```

To upgrade the device firmware, complete the steps listed for the selected method in the :doc:`matter:nrfconnect_examples_software_update` tutorial of the Matter documentation.

.. _matter_lock_snippets:

> Snippets

Reviewer: Snippets section should be under Configuration, after the custom configurations.

========

.. |snippet| replace:: :makevar:`lock_SNIPPET`

.. include:: /includes/sample_snippets.txt

The following snippet is available:

* ``schedules`` - Enables schedule timed access features.

>   .. note::

Reviewer: I notice we've got the same note in the Light switch sample. If we expect to need the note more often, we should probably make it into an include file that we can reuse.

Author: I've created a .txt file for that :)

    This snippet requires a custom ZAP configuration, located in the :file:`snippets/schedules` directory.

>     Ensure to configure and generate the correct ZAP files if a custom data model is needed.

Reviewer: ```suggestion
    If you need a custom data model, you must configure and generate the correct ZAP files.
```

    To do this, run the ``west zap-generate`` command and choose the :file:`snippets/schedules/lock.zap` file.

.. _matter_lock_enabling_scheduled_timed_access:

> Enabling scheduled timed access

Reviewer: I don't think this section is needed anymore, seeing as enabling the snippet is so straightforward.

Just move the information after the example command to the section describing the feature, so no information is lost.

===============================

To enable the scheduled timed access feature use the ``schedules`` snippet available for the :ref:`Matter Lock sample <matter_lock_snippets>`.
To use the snippet and enable schedule timed access use the ``lock_SNIPPET=schedules`` build flag.

For example:

.. code-block:: console

>    west build -b nrf52840dk/nrf52840 -p -- -Dlock_SNIPPET=schedules

Reviewer 2: I think that this part might be redundant as it will be visible under `matter_lock_snippets` after including `.. include:: /includes/sample_snippets.txt` :
```
Specify the corresponding snippet names in the |SNIPPET| CMake option.
```
So IMO the line above might be enough here, thoughts?

Author: I've removed it :)

.. note::
   You can test :ref:`matter_lock_scheduled_timed_access` using any Matter compatible controller.

>    The following steps use the CHIP Tool controller as an example.
>    All scheduled timed access entries are saved to non-volatile memory and loaded automatically after device reboot.

Reviewer: ```suggestion
   The following steps use the CHIP Tool controller as an example.

   All scheduled timed access entries are saved to non-volatile memory and loaded automatically after device reboot.
```
